packages/vscode-extension/src/builders/buildIOS.ts

```diff
@@ -199,6 +203,39 @@ export async function buildIos(
   }
 }
 
+async function getOrMakeSimulator(buildConfig: IOSLocalBuildConfig) {
+  const { runtimeId } = buildConfig;
+  const devices = await listSimulators(SimulatorDeviceSet.Default); // use default location becuase xcodebuild can't use custom device sets
+  const matchingDevice = devices.find(
+    (deviceInfo) => deviceInfo.runtimeInfo?.identifier === runtimeId && deviceInfo.available
+  );
+  if (matchingDevice) {
+    Logger.info("Using existing simulator for iOS build:", matchingDevice);
+    return { simulatorUdid: matchingDevice.UDID };
+  }
+  Logger.info("Creating new simulator for iOS build:", runtimeId);
+  // It is unlikely we will reach this point as it may only happen when user doesn't have simulator for the selected runtime available
+  // The above is possible but when Xcode installs new SDK, it also creates a set of simulators for it
+  // Thereofre, this can only happen when the user has deleted all of the simulator or if they deleted the runtime
+  // in which case we won't be able to create a new simulator for it anyways and the build should fail.
+  const simulatorUdid = await createSimulatorWithRuntimeId(
+    "iPhone 16 Pro", // device type doesn't matter as long as it supports the runtime, we choose iPhone 16 as it is recent and will be available for a while
+    "Radon IDE Temp Simulator",
+    runtimeId,
+    SimulatorDeviceSet.Default
+  );
```

Comment:
How long does this take? I wouldn't expect this to impact build times much, but maybe it'd be better to create this temp device once (when it's first needed), and then remove it once the whole extension is shutting down, to avoid a situation where the device is created and deleted over and over when a user rebuilds the app?

Reply:
on my mac M2 it takes consistently below 100ms